openupgrade_scripts/scripts/website_sale/14.0.1.0/post-migration.py

```diff
@@ -1,10 +1,38 @@
 # Copyright (C) 2021 Open Source Integrators <https://www.opensourceintegrators.com/>
 # Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import re
 
 from openupgradelib import openupgrade
 
 
+def extract_custom_product_page_term_conditions(env):
+    """Replace Terms and Conditions content in the new v14 template so as not to lose
+    content from previous versions if it has been customised."""
+    product_custom_text_view = env.ref("website_sale.product_custom_text")
+    if not product_custom_text_view:
+        return
+    product_custom_text_arch = product_custom_text_view.arch_db
+    product_custom_text_pattern = r'<p class="text-muted(.*?)<\/p>'
+    product_custom_text_matches = re.findall(
+        product_custom_text_pattern, product_custom_text_arch, re.DOTALL
+    )
+    product_views = env["ir.ui.view"].search(
+        [("key", "=", "website_sale.product"), ("website_id", "!=", False)]
+    )
+    for view in product_views:
+        product_arch = view.arch_db
+        product_pattern = r'<hr />\n<p class="text-muted">(.*?)<\/p>'
```

Comment:
Are you sure this is going to do a match? Blank spaces count as characters as well :)

Reply:
Changed and tested in the old template that performs the search correctly with this pattern, thank you.